File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.py
def solution(arr):
    answer = []
    check_number = arr[0]
    for i in range(len(arr)):

        # 2. 이전 숫자와 중복이라면 패스
        # 이전 숫자와 같거나, 처음 값이 아니라면
        if (arr[i] == check_number) and (i != 0):
            continue

        # 1. 반복문을 순회하며 처음보는 숫자는 배열에 저장.
        answer.append(arr[i])

        # 3. 중복이 아니라면 추가하고 체크하는 숫자를 변경
        check_number = arr[i]

    return answer

File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.java
import java.util.*;

public class Solution {
    public int[] solution(int []arr) {
        List<Integer> numberList = new ArrayList<>();

        int checkNumber = arr[0];

        // 연속된 숫자일 경우 제거하고 반환하기
        for(int i = 0; i < arr.length; i++){

            if(arr[i] == checkNumber && i != 0){
                continue;
            }

            numberList.add(arr[i]);
            checkNumber = arr[i];

        }

        int[] answer = new int[numberList.size()];
        for(int i = 0; i < numberList.size(); i++){
            answer[i] = numberList.get(i);
        }

        return answer;
    }
}